src/Functions/FunctionSQLJSON.h

```diff
@@ -147,43 +157,95 @@ class FunctionSQLJSONHelpers
+                {
+                    for (const auto& element_type : tuple_type->getElements())
+                    {
+                        if (!isString(element_type))
+                        {
+                            throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT, "JSONPath functions require second argument to be JSONPath of type Tuple(String)");
+                        }
+                    }
+                }
+                if (const auto * array_type = checkAndGetDataType<DataTypeArray>(json_path_column.type.get()))
+                {
+                    if (!isString(array_type->getNestedType()))
+                    {
+                        throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT, "JSONPath functions require second argument to be JSONPath of type Array(String)");
+                    }
+                }
             }
             if (!isColumnConst(*json_path_column.column))
             {
-                throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT, "Second argument (JSONPath) must be constant string");
+                throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT, "Second argument (JSONPath) must be constant string/tuple/array");
             }
-
+            auto parse_json_path = [&](const String& query) -> std::shared_ptr<GeneratorJSONPath<JSONParser>>
+            {
+                /// Tokenize the query
+                Tokens tokens(query.data(), query.data() + query.size());
+                /// Max depth 0 indicates that depth is not limited
+                IParser::Pos token_iterator(tokens, parse_depth, parse_backtracks);
+                /// Parse query and create AST tree
+                Expected expected;
+                ASTPtr res;
+                ParserJSONPath parser;
+                const bool parse_res = parser.parse(token_iterator, res, expected);
+                if (!parse_res)
+                {
+                    throw Exception(ErrorCodes::BAD_ARGUMENTS, "Unable to parse JSONPath");
+                }
+                return std::make_shared<GeneratorJSONPath<JSONParser>>(res);
+            };
             /// Prepare to parse 1 argument (JSONPath)
-            String query = typeid_cast<const ColumnConst &>(*json_path_column.column).getValue<String>();
-
-            /// Tokenize the query
-            Tokens tokens(query.data(), query.data() + query.size());
-            /// Max depth 0 indicates that depth is not limited
-            IParser::Pos token_iterator(tokens, parse_depth, parse_backtracks);
-
-            /// Parse query and create AST tree
-            Expected expected;
-            ASTPtr res;
-            ParserJSONPath parser;
-            const bool parse_res = parser.parse(token_iterator, res, expected);
-            if (!parse_res)
+            std::vector<std::shared_ptr<GeneratorJSONPath<JSONParser>>> generator_json_paths;
+            std::vector<bool> path_asterisks;
```

Comment:
Is it part of another change? Let's remove it from here and then have another pull request.

Reply:
Yes, it's about to solve the problem when json path contains `[*]`, and I have removed it now and I will open another pr for it.